### ISSUE TITLE
Easy way to loop over and filter pages inside component.

### ISSUE THREAD

**Issue by thom801** [nuxt/nuxt.js#6840]
### What problem does this feature solve?

Users of Nuxt will often need a way to dynamically generate navigation items in the template of a Vue Component. There would need to be a way for adding things like page title etc to a page for display purposes as well.

### What does the proposed changes look like?

// /pages/blog.vue

<template lang="html">
  <ul>
    <li v-for="post in pages.blog">
      <a href="{{ post.path }}">{{ post.title }}</a>
    </li>
  </ul>
</template>

// /pages/blog/my-blog-post.vue

<script>
export default {
  extendPage: {
    title: "My Blog Post",
    published_date: "Jan 1, 2020",
  }
}
</script>

Something like that, but I'm totally open to better solutions.

<div align="right"><sub><em>This feature request is available on <a href="https://cmty.app/nuxt">Nuxt</a> community (<a href="https://cmty.app/nuxt/nuxt.js/issues/c10175">#c10175</a>)</em></sub></div>

**Comment by talebi1** [comment]
why not use the head method https://nuxtjs.org/api/pages-head/ ?

**Comment by thom801** [comment]
@talebi1 how would that solve this problem?

**Comment by ncolletti** [comment]
I agree with @talebi1 assessment, unless we are misunderstanding the request. You can add the dynamic content to the head function on your component page to set the title or other page descriptions needed. Example from what you provided:

/pages/blog/my-blog-post.vue

`<template>
  <h1>{{ title }}</h1>
</template>

<script>
export default {
  data () {
    return {
      title: 'My Blog Post',
      published_date: 'Jan 1, 2020'
    }
  },
  head () {
    return {
// sets page title and other meta data as needed for component/blog-post
      title: this.title,
      meta: [
        { hid: 'description', published_date: this.published_date content: 'My custom description' }
      ]
    }
  }
}
</script>`

**Comment by talebi1** [comment]
@thom801  created a little example here to demonstrate my point 
https://codesandbox.io/s/cool-dewdney-gild9

**Comment by thom801** [comment]
@ncolletti I think you are misunderstanding because I need a way to easily access an array of pages or posts and then display them on a page. What you did attaches the title, but then how do you access that title from another template **easily**

@talebi1 thank you for the example but I think you just wrote a bunch of semi-complex code to pull in the posts manually. What I'm saying is there should be a very simple way of accessing those pages/posts without writing a big block of code like this in each component:

```
this.$router.options.routes.forEach(r => {
      if (r.path.startsWith(this.$route.path + "blog")) {
        r.component().then(function(blog) {
          let title = blog.data().title;
          self.pages.push({
            path: r.path,
            name: title
          });
        });
      }
    });
```

Instead of having that method you should just have a global `pages` and `posts` variable that you can use in any template or something like that. To be clear, I'm not saying it's impossible to do this right now, I am proposing a new feature to make it very easy to do.

**Comment by talebi1** [comment]
@thom801 I agree. but I think you can make your like more easier if you can pull the pages/blogs data from an api endpoint then render then with a shared template.

**Comment by Triloworld** [comment]
Just Vuex or simply emit (example):

```
/pages/blog.vue

<template lang="html">
  <ul>
    <li v-for="post in pages.blog">
      <a href="{{ post.path }}">{{ post.title }}</a>
    </li>
  </ul>
</template>
<script>
export default {
    this.$root.$on("postHead", data => {
      if (data.title != '') {
        this.head.title = data.head;
      }
    });
}
</script>
```
```
/pages/blog/my-blog-post.vue

<script>
export default {
  head: {
    title: "My Blog Post",
    published_date: "Jan 1, 2020",
  },
  mounted() {
     this.$root.$emit("postHead", {
        head: this.head, //Maybe different path to access but must be available
     });
  },
}
</script>
```
I use in different scenario. Homepage is type of "One Page" and every section emit id and title. In head I generate from It menu to navigate without duplicate and removing it after unmount.
